bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/HeapSizeManager.java

```diff
@@ -183,5 +189,10 @@ public void onFailure(Throwable t) {
 
   public void markCanBeCompleted(Long id) {
     completedOperationIds.offer(id);
+    if (this.isFlushing.get() && completedOperationIds.size() == pendingOperationsWithSize.size()) {
```

Comment:
sorry! can you explain a bit on why isFlushing has to be true and "ompletedOperationIds.size() == pendingOperationsWithSize.size()", so that we should notify thread waiting on "this"?